CHAOS-6087 Linux API status code (#7358)
harness/developer-hub: docs/chaos-engineering/chaos-faults/linux/linux-api-status-code.md

---
id: linux-api-status-code
title: Linux API status code
---

import Ossupport from './shared/note-supported-os.md'
import FaultPermissions from './shared/fault-permissions.md'

> Linux API status code changes the API response status code and optionally API  response body through path filtering. This is achieved by starting the proxy server and redirecting the traffic through the proxy server.

Reviewer: ```suggestion
Linux API status code changes the API response status code and optionally the API response body through path filtering. This is achieved by starting the proxy server and redirecting the traffic through the proxy server.
```

![Linux API status code](./static/images/linux-api-status-code.png)

## Use cases
Linux API status code:

> - Tests the error handling capabilities of API and client applications. By changing the API response status code to different error codes, such as 400 (Bad Request) or 500 (Internal Server Error), you can evaluate how well your application handles and responds to various error scenarios.
> - Simulates situations where the API may be temporarily unavailable or rate-limited by returning temporary error codes like 503 (Service Unavailable) or 429 (Too Many Requests).
> - Used for content filtering, by selectively filter or block certain responses. For example, you can change the status code to 404 (Not Found) for specific paths or patterns, indicating that the requested resource does not exist.

Reviewer: ```suggestion
- Tests the error handling capabilities of API and client applications. By changing the API response status code to different error codes, such as 400 (Bad Request) or 500 (Internal Server Error), you can evaluate how well your application handles and responds to various error scenarios.
- Simulates situations where the API may be temporarily unavailable or rate-limited by returning temporary error codes like 503 (Service Unavailable) or 429 (Too Many Requests).
- Used for content filtering, by selectively filtering or blocking certain responses. For example, you can change the status code to 404 (Not Found) for specific paths or patterns, indicating that the requested resource does not exist.
```

      <td> For more information, go to <a href="#advanced-filters">source IPs</a>.</td>
  </tr>
  <tr>
      <td> destinationHosts </td>
      <td> Comma-separated destination host names are used as filters, indicating the hosts on which you call the API. This specification applies exclusively to the "egress" type. </td>
      <td> For more information, go to <a href="#advanced-filters">destination hosts</a>.</td>
  </tr>
  <tr>
    <td> destinationIPs </td>
    <td> Comma-separated destination IPs are used as filters, indicating the hosts on which you call the API. This specification applies exclusively to the "egress" type.</td>
    <td> For more information, go to <a href="#advanced-filters">destination hosts</a>.</td>
  </tr>
  <tr>
    <td> responseBody </td>
    <td>  String body to overwrite the HTTP response body.</td>
    <td> If no value is provided, response will be an empty body. Default is an empty body. For more information, go to <a href="#advanced-fault-tunables">response body </a></td>
  </tr>
  <tr>
      <td> statusCode </td>

>       <td> Status code received when the API is blocked. </td>
>       <td> When the API is blocked, it throws an error along with the status code. Default: 404. </td>

Reviewer: The API isn't blocked in this fault, it will just receive a custom status code.
```suggestion
      <td> Response status code received when the API request is sent. </td>
      <td> Default: 404. </td>
```
